diff --git a/typing.d.ts b/typing.d.ts
index 6ec4ac719ff113183076239d4044c323da02628c..612a4ec62f956f0abe5863c9201e5ec6e8827273 100644
--- a/typing.d.ts
+++ b/typing.d.ts
@@ -8,7 +8,7 @@ declare module 'd3-svg-to-png' {
     cssinline: number
     background: string
   }
-  function d3ToPng(selector: string, fileName: string, options?: Partial<Options>): Promise<string>
+  function d3ToPng(selectorOrElement: string | Element, fileName: string, options?: Partial<Options>): Promise<string>
 
   export default d3ToPng
 }
