diff --git a/obsidian.d.ts b/obsidian.d.ts
index caeee06ea2205845837d6155c88672d8ac8ce400..fa294f736ead09e823c6c40810d6b7b2375f43a8 100644
--- a/obsidian.d.ts
+++ b/obsidian.d.ts
@@ -1676,7 +1676,7 @@ export function displayTooltip(newTargetEl: HTMLElement, content: string | Docum
  * @public
  * @since 0.9.7
  */
-export class DropdownComponent extends ValueComponent<string> {
+export class DropdownComponent<T extends string = string> extends ValueComponent<T> {
     /**
      * @public
      * @since 0.9.7
@@ -1696,27 +1696,27 @@ export class DropdownComponent extends ValueComponent<string> {
      * @public
      * @since 0.9.7
      */
-    addOption(value: string, display: string): this;
+    addOption(value: T, display: string): this;
     /**
      * @public
      * @since 0.9.7
      */
-    addOptions(options: Record<string, string>): this;
+    addOptions(options: Record<T, string>): this;
     /**
      * @public
      * @since 0.9.7
      */
-    getValue(): string;
+    getValue(): T;
     /**
      * @public
      * @since 0.9.7
      */
-    setValue(value: string): this;
+    setValue(value: T): this;
     /**
      * @public
      * @since 0.9.7
      */
-    onChange(callback: (value: string) => any): this;
+    onChange(callback: (value: T) => any): this;
 }
 
 /**
@@ -5068,7 +5068,7 @@ export class Setting {
     /**
      * @public
      */
-    constructor(containerEl: HTMLElement);
+    constructor(containerEl: Node);
     /**
      * @public
      * @since 0.12.16
@@ -5139,7 +5139,7 @@ export class Setting {
      * @public
      * @ince 0.9.7
      */
-    addDropdown(cb: (component: DropdownComponent) => any): this;
+    addDropdown<T extends string>(cb: (component: DropdownComponent<T>) => void): this;
     /**
      * @public
      * @ince 0.16.0
