diff --git a/obsidian.d.ts b/obsidian.d.ts
index 73bb2491f416649b1772cbd04b68ed9433707dab..9877b563ccd49b4b74e76adce3758011d9895664 100644
--- a/obsidian.d.ts
+++ b/obsidian.d.ts
@@ -1702,7 +1702,7 @@ export function displayTooltip(newTargetEl: HTMLElement, content: string | Docum
  * @public
  * @since 0.9.7
  */
-export class DropdownComponent extends ValueComponent<string> {
+export class DropdownComponent<T extends string = string> extends ValueComponent<T> {
     /**
      * @public
      * @since 0.9.7
@@ -1722,27 +1722,27 @@ export class DropdownComponent extends ValueComponent<string> {
      * @public
      * @since 0.9.7
      */
-    addOption(value: string, display: string): this;
+    addOption(value: T, display: string): this;
     /**
      * @public
      * @since 0.9.7
      */
-    addOptions(options: Record<string, string>): this;
+    addOptions(options: Record<T, string>): this;
     /**
      * @public
      * @since 0.9.7
      */
-    getValue(): string;
+    getValue(): T;
     /**
      * @public
      * @since 0.9.7
      */
-    setValue(value: string): this;
+    setValue(value: T): this;
     /**
      * @public
      * @since 0.9.7
      */
-    onChange(callback: (value: string) => any): this;
+    onChange(callback: (value: T) => any): this;
 }
 
 /**
@@ -5182,7 +5182,7 @@ export class Setting {
     /**
      * @public
      */
-    constructor(containerEl: HTMLElement);
+    constructor(containerEl: Node);
     /**
      * @public
      * @since 0.12.16
@@ -5253,7 +5253,7 @@ export class Setting {
      * @public
      * @ince 0.9.7
      */
-    addDropdown(cb: (component: DropdownComponent) => any): this;
+    addDropdown<T extends string>(cb: (component: DropdownComponent<T>) => void): this;
     /**
      * @public
      * @ince 0.16.0
